contracts/experimental/ERC20ProxyForZRC2/test/basicProxyTest.ts

```diff
@@ -0,0 +1,179 @@
+    proxyDeployer = new ethers.Wallet(process.env.TEST_KEY_1, ethers.provider);
+    tokenHolder = new ethers.Wallet(process.env.TEST_KEY_2, ethers.provider);
+    console.log(`ChainId ${hre.network.config.chainId & 0x7fff}`);
+    initZilliqa(
+      hre.network.config.url,
+      hre.network.config.chainId & 0x7fff,
+      hre.network.config.accounts,
+      CONFIRM_ATTEMPTS,
+    );
+    zrc2Owner = hre.zilliqa.getDefaultAccount();
+    //zrc2OwnerAddress = zrc2Owner.address.toLowerCase();
+
+    if (!process.env.CACHED) {
+      zrc2Contract = await hre.deployScillaContract(
+        "FungibleToken",
+        zrc2OwnerEVM.address,
+        ZRC2_NAME,
+        ZRC2_SYMBOL,
+        ZRC2_DECIMALS,
+        ZRC2_SUPPLY,
+      );
+      console.log(
+        `Sample ZRC-2 token ${zrc2Contract.address} owned by ${zrc2Owner.address}`,
+      );
+      const erc20Factory = (
+        await ethers.getContractFactory("ZRC2ERC20Proxy")
+      ).connect(proxyDeployer);
+      erc20Proxy = await erc20Factory.deploy(
+        zrc2Contract.address.toLowerCase(),
+      );
+      await erc20Proxy.waitForDeployment();
+    } else {
+      zrc2Contract = await hre.interactWithScillaContract(
+        "0x178ABcED2552522F131E7C89E80E622862E6c00E",
+      );
+      erc20Proxy = (await ethers.getContractFactory("ZRC2ERC20Proxy"))
+        .connect(proxyDeployer)
+        .attach("0x0C9fb168f7155Ea54aAaCafdbD9A652bd895b4a4");
+    }
+    console.log(`ERC20 proxy deployed at ${erc20Proxy.target}`);
+    console.log(` .... proxying to ZRC2 at ${await erc20Proxy.zrc2_proxy()}`);
+    console.log(
+      `Proxy deployer ${proxyDeployer.address} ; token holder ${tokenHolder.address}`,
+    );
+  });
+
+  it("0000 Should deploy successfully", async function () {
+    expect(zrc2Contract.address).to.be.properAddress;
+    expect(erc20Proxy.target).to.be.properAddress;
+  });
+
+  it("0001 Should report parameters correctly", async function () {
+    expect(await erc20Proxy.decimals()).to.equal(ZRC2_DECIMALS);
+    expect(await erc20Proxy.symbol()).to.equal(ZRC2_SYMBOL);
+    expect(await erc20Proxy.name()).to.equal(ZRC2_NAME);
+    expect(await erc20Proxy.totalSupply()).to.equal(ZRC2_SUPPLY);
+    expect((await erc20Proxy.zrc2_proxy()).toLowerCase()).to.equal(
+      zrc2Contract.address.toLowerCase(),
+    );
+  });
+
+  it("0002 Should deal with transfers correctly", async function () {
+    const AMT = 540;
+    expect(await erc20Proxy.balanceOf(zrc2OwnerEVM.address)).to.equal(
+      ZRC2_SUPPLY,
+    );
+    expect(await erc20Proxy.balanceOf(tokenHolder.address)).to.equal(0);
+    await (
+      await erc20Proxy.connect(zrc2OwnerEVM).transfer(tokenHolder, AMT)
+    ).wait();
+    expect(await erc20Proxy.balanceOf(tokenHolder.address)).to.equal(AMT);
+    expect(await erc20Proxy.balanceOf(zrc2OwnerEVM.address)).to.equal(
+      ZRC2_SUPPLY - AMT,
+    );
+    await (
+      await erc20Proxy.connect(tokenHolder).transfer(zrc2OwnerEVM, AMT)
```

Comment:
The transfer code could use a little bit of cleanup. What you really want here is https://ethereum-waffle.readthedocs.io/en/latest/matchers.html?highlight=changeBalance#change-token-balance-multiple-accounts, but I guess it won't work given that `not.to.be.reverted` fails due to issue with receipt handling.
I propose the following wrapper
```
async function assertedTransfer(proxy: ZRC2ERC20Proxy, from: Wallet, to: Wallet, amt: number) {
    const receipt = await (
      await proxy.connect(from).transfer(to, amt)
    ).wait();

    expect(receipt?.status === 1, `Transfer of ${amt} from ${from.address} to ${to.address} failed. Transaction status ${receipt?.status || "unknown"}`);
  }
  ```
  So all the transfer lines become
  ```
  await assertedTransfer(erc20Proxy, tokenHolder, zrc2OwnerEVM, AMT);
  ```
  Benefit of that is that the transfer call is actually asserted and gives back some info about error

Reply:
Done